Add default security constraint to FGP.19139.HNAP template.

wangf1122 opened the pull request:
This is for FGP import template only. It will not impact other import jobs.

Security constraint is added if the xml metadata is lack of. Otherwise, leave as is. The overwrite existing security constraint to default code is added as well but it's been disabled. In the future, we can enable it if it's needed.

Commit:
Add default security constraint to FGP.19139.HNAP template.

ianwallen commented on src/main/config/conversion/import/iso19139.ca.HNAP.FGP-to-iso19139.ca.HNAP.xsl:
```diff
@@ -102,6 +102,41 @@
+    </xsl:copy>
+  </xsl:template>
+
+  <!--Overwrite security constraint to default unclassified if presented in metadata xml-->
```
Please remove comment as we are not overwriting existing classification values.  If it fails to validate user will have to change manually modify the value.

The only thing that we may overwrite is if the user note is using an old value then we may want to change it to the correct value.  Otherwise the user would have to change that as well.
See https://github.com/metadata101/iso19139.ca.HNAP/pull/195
i.e. "Protect A" could be changed to "Protected A"

wangf1122 replied:
Removed.

ianwallen commented on src/main/config/conversion/import/iso19139.ca.HNAP.FGP-to-iso19139.ca.HNAP.xsl:
```diff
@@ -102,6 +102,41 @@
+        <xsl:when test="gmd:resourceConstraints/gmd:MD_SecurityConstraints/gmd:classification/gmd:MD_ClassificationCode">
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:element name="gmd:resourceConstraints">
```
I believe it would be cleaner to simply add the element instead of using xsl:element/xsl:attribute/xsl:text.
i.e. 
https://github.com/metadata101/iso19139.ca.HNAP/blob/16537ee38e8492d1fb5d971f150e0016357cb46e/src/main/plugin/iso19139.ca.HNAP/update-fixed-info.xsl#L239-L250

ianwallen commented on src/main/config/conversion/import/iso19139.ca.HNAP.FGP-to-iso19139.ca.HNAP.xsl:
```diff
@@ -102,6 +102,41 @@
+  <xsl:template match="gmd:MD_DataIdentification">
+    <xsl:copy copy-namespaces="no">
+      <xsl:choose>
+        <xsl:when test="gmd:resourceConstraints/gmd:MD_SecurityConstraints/gmd:classification/gmd:MD_ClassificationCode">
```
May want the try something like the following
`<xsl:if test="not(gmd:resourceConstraints/gmd:MD_SecurityConstraints/gmd:classification/gmd:MD_ClassificationCode)">`
instead of creating an empty condition.

wangf1122 replied:
Good suggestion. I will try.

Commit:
Remove overwriting security constraint in FGP template.

Commit:
Reorganize xsl code.

wangf1122 commented:
@ianwallen 

I made some clean up to the code. 

1) Removed the Choose and replaced with a simple "if" statement.
2) Put xml element directly without using <xsl:element>
